app/controllers/fandoms_controller.rb

```diff
@@ -5,22 +5,22 @@ def index
     if @collection
       @media = Media.canonical.by_name - [Media.find_by(name: ArchiveConfig.MEDIA_NO_TAG_NAME)] - [Media.find_by(name: ArchiveConfig.MEDIA_UNCATEGORIZED_NAME)]
       @page_subtitle = @collection.title
-      @medium = Media.find_by_name(params[:medium_id]) if params[:medium_id]
+      @medium = Media.find_by_name(params[:media_id]) if params[:media_id]
       @counts = SearchCounts.fandom_ids_for_collection(@collection)
       @fandoms = (@medium ? @medium.fandoms : Fandom.all).where(id: @counts.keys).by_name
-    elsif params[:medium_id]
-      if @medium = Media.find_by_name(params[:medium_id])
+    elsif params[:media_id]
+      if @medium = Media.find_by_name(params[:media_id])
```

Comment:
Let's do a `find_by!` and avoid the if-else-raise `ActiveRecord::RecordNotFound` setup entirely.